### PR TITLE
PULSE-192

### PR DESCRIPTION
**Pull request by Namanj**
https://phdata.atlassian.net/secure/RapidBoard.jspa?rapidView=4&projectKey=PULSE&modal=detail&selectedIssue=PULSE-192

This isn't working in my unit tests yet.

### CODE REVIEW

**Comment by Namanj** [comment]
I think I've made all the required updates to automatically include the hostname in the Python appender, but I'm not seeing the "hostName" field show up in the Solr collection schema. 

I also tested the atomic piece of logic update from "appenders/python/pulse_appender/LogFormatter.py" on the Valhalla edge node edge1, and it's working as expected.

File: cloudera-integration/parcel/conf/solr-configs/pulseconfigv2/schema.xml [hunk 1]
```diff
@@ -40,6 +40,8 @@
 
         <field name="text" type="text_general" indexed="true" stored="false" multiValued="true"/>
 
+        <field name="hostName" type="text_general" indexed="true" stored="true" multiValued="false"/>
```
Comment (afoerster) [inline cloudera-integration/parcel/conf/solr-configs/pulseconfigv2/schema.xml]: use lower case 'hostname' here.

File: cloudera-integration/parcel/conf/solr-configs/pulseconfigv2/schema.xml [hunk 2]
```diff
@@ -64,6 +66,7 @@
     <copyField source="message" dest="text"/>
     <copyField source="threadName" dest="text"/>
     <copyField source="throwable" dest="text"/>
+    <copyField source="hostName" dest="text"/>
```
Comment (afoerster) [inline cloudera-integration/parcel/conf/solr-configs/pulseconfigv2/schema.xml]: lowercase

File: appenders/python/pulse_appender/LogFormatter.py [hunk 3]
```diff
@@ -24,4 +25,6 @@ def format(self, record):
         data["level"] = record.levelname
         data["message"] = record.msg
         data["threadName"] = record.threadName
+        data["hostName"] = socket.gethostname()
```
Comment (afoerster) [inline appenders/python/pulse_appender/LogFormatter.py]: Does this always work?

Comment (afoerster) [reply appenders/python/pulse_appender/LogFormatter.py]: Should we wrap it in a try/catch?

File: common/src/main/scala/io/phdata/pulse/common/domain/LogEvent.scala [hunk 4]
```diff
@@ -17,21 +17,23 @@
...
+  * @param properties MDC properties
+  */
 case class LogEvent(id: Option[String],
                     category: String,
                     timestamp: String,
                     level: String,
                     message: String,
                     threadName: String,
                     throwable: Option[String] = None,
+                    hostname: String,
```
Comment (afoerster) [inline common/src/main/scala/io/phdata/pulse/common/domain/LogEvent.scala]: You'll have to make this value optional so we can keep backwards compatibility (like `throwable`)

File: common/src/main/scala/io/phdata/pulse/common/DocumentConversion.scala [hunk 5]
```diff
@@ -17,11 +17,11 @@
 package io.phdata.pulse.common
 
 import io.phdata.pulse.common.domain.LogEvent
-import org.apache.solr.common.{ SolrDocument, SolrInputDocument }
+import org.apache.solr.common.{SolrDocument, SolrInputDocument}
 
 /**
- *
- */
+  *
```
Comment (afoerster) [inline common/src/main/scala/io/phdata/pulse/common/DocumentConversion.scala]: Empty comment?

**Comment by afoerster** [comment]
> I think I've made all the required updates to automatically include the hostname in the Python appender, but I'm not seeing the "hostName" field show up in the Solr collection schema.
Make sure you are creating a new collection, otherwise the hostname field won't show up unless you leave the Collection Roller running

File: common/src/main/scala/io/phdata/pulse/common/DocumentConversion.scala (hunk 5 above)
Comment (Namanj) [reply common/src/main/scala/io/phdata/pulse/common/DocumentConversion.scala]: I think this was just the result of reformatting the code, it wasn't deliberate

Comment (afoerster) [reply common/src/main/scala/io/phdata/pulse/common/DocumentConversion.scala]: I see. Yes you can remove it.

**Comment by Namanj** [comment]
@afoerster I've made all the changes I can think of, but I haven't been able to get this to work on Valhala yet.

I get this error in the log-collector when I run my spark log producing app:
org.apache.solr.client.solrj.impl.HttpSolrServer$RemoteSolrException: ERROR: [doc=e7a1e838-3f06-490f-88b6-a11d74f2ed3b] multiple values encountered for non multiValued field hostname: [, edge1.valhalla.phdata.io]

Not sure why & where this field is becoming a list. I'll keep digging tomorrow.

**Comment by Namanj** [comment]
>Make sure you are creating a new collection, otherwise the hostname field won't show up unless you leave the Collection Roller running

How do I ensure this? Currently I scp all my updates to the edge node, make package, and then start all 3 services along with the spark app.

My newly created solr collection doesn't have the 'hostname' field

**Comment by afoerster** [comment]
@Namanj you can try creating a collection manually using your config with the solrctl command: https://www.cloudera.com/documentation/enterprise/5-14-x/topics/search_solrctl_examples.html

**Pull request closed without merging**